shared/texture.h

```diff
@@ -0,0 +1,18 @@
+#ifndef ENIGMA_SHARED_TEXTURE_H
+#define ENIGMA_SHARED_TEXTURE_H
+
+namespace enigma {
+struct Texture {
+  Texture() : width(0), height(0), channels(1), ID(-1), pxdata(nullptr) {}
+  Texture(unsigned width, unsigned height, unsigned char* pxdata, unsigned channels) : width(width), height(height), channels(channels), pxdata(pxdata) {}
+  Texture(unsigned width, unsigned height, int ID) : width(width), height(height), channels(1), ID(ID) {}
+  void init(); // loads a texture and deletes pxdata (engine only)
+  unsigned width;
+  unsigned height;
+  unsigned channels;
+  int ID;
+  unsigned char* pxdata;
+};
+} //namespace enigma
+
+#endif
```

Comment:
Uhm, I don't know how I feel about this. This is confusing because graphics texture vs raw image class you added. Can we not just use raw image here? Too bad we can't use our protos in the engine yet.

Reply:
This is used to load a texture later. I wrote it long before RawImage you could modify RawImage to have a load later I guess